Description: Block manual building in the DKMS source tree.
  To avoid messing up future DKMS builds and the zfs installation,
  block manual building of the DKMS source tree.
Author: unknown
Origin: ubuntu
Forwarded: not-needed
Reviewed-By: Petter Reinholdtsen <pere@hungry.com>
Last-Update: 2017-10-06

Index: zfs-linux/config/dkms.m4
===================================================================
--- /dev/null
+++ zfs-linux/config/dkms.m4
@@ -0,0 +1,14 @@
+dnl #
+dnl # Prevent manual building in DKMS source tree.
+dnl #
+AC_DEFUN([ZFS_AC_DKMS_INHIBIT], [
+	AC_MSG_CHECKING([for dkms.conf file])
+        AS_IF([test -e dkms.conf], [
+		AC_MSG_ERROR([
+	*** ZFS should not be manually built in the DKMS source tree.
+	*** Remove all ZFS packages before compiling the ZoL sources.
+	*** Running "make install" breaks ZFS packages.])
+        ], [
+		AC_MSG_RESULT([not found])
+        ])
+])
Index: zfs-linux/config/user.m4
===================================================================
--- zfs-linux.orig/config/user.m4
+++ zfs-linux/config/user.m4
@@ -2,6 +2,7 @@ dnl #
 dnl # Default ZFS user configuration
 dnl #
 AC_DEFUN([ZFS_AC_CONFIG_USER], [
+	ZFS_AC_DKMS_INHIBIT
 	ZFS_AC_CONFIG_USER_GETTEXT
 	ZFS_AC_CONFIG_USER_MOUNT_HELPER
 	ZFS_AC_CONFIG_USER_SYSVINIT
